tests/callbacks/test_callbacks.py

```diff
@@ -51,6 +53,46 @@ def test_trainer_callback_system(torch_save):
 
     trainer.fit(model)
 
+    # check how many times a method was called
+    assert callback_mock.on_init_start.call_count == 1
+    assert callback_mock.on_init_end.call_count == 1
+    assert callback_mock.setup.call_count == 1
+    assert callback_mock.on_fit_start.call_count == 1
+    assert callback_mock.on_pretrain_routine_start.call_count == 1
+    assert callback_mock.on_pretrain_routine_end.call_count == 1
+    assert callback_mock.on_sanity_check_start.call_count == 1
+    assert callback_mock.on_validation_start.call_count == limit_val_batches * 2
+    assert callback_mock.on_validation_epoch_start.call_count == limit_val_batches * 2
+    assert callback_mock.on_validation_batch_start.call_count == limit_val_batches * 2
+    assert callback_mock.on_validation_batch_end.call_count == limit_val_batches * 2
+    assert callback_mock.on_validation_epoch_end.call_count == limit_val_batches * 2
+    assert callback_mock.on_validation_end.call_count == limit_val_batches * 2
+    assert callback_mock.on_sanity_check_end.call_count == 1
+    assert callback_mock.on_train_start.call_count == 1
+    assert callback_mock.on_epoch_start.call_count == 1
+    assert callback_mock.on_train_epoch_start.call_count == 1
+    assert callback_mock.on_batch_start.call_count == limit_train_batches
+    assert callback_mock.on_train_batch_start.call_count == limit_train_batches
+    assert callback_mock.on_after_backward.call_count == limit_train_batches
+    assert callback_mock.on_before_zero_grad.call_count == limit_train_batches
+    assert callback_mock.on_batch_end.call_count == limit_train_batches
+    assert callback_mock.on_train_batch_end.call_count == limit_train_batches
+    assert callback_mock.on_save_checkpoint.call_count == 1
+    assert callback_mock.on_epoch_end.call_count == 1
+    assert callback_mock.on_train_epoch_end.call_count == 1
+    assert callback_mock.on_train_end.call_count == 1
+    assert callback_mock.on_fit_end.call_count == 1
+    assert callback_mock.teardown.call_count == 1
```

Comment:
use mock with explicit calls, see
https://github.com/PyTorchLightning/pytorch-lightning/blob/0b653b89baf1a36623532d93cb646679d4c8c080/tests/callbacks/test_callbacks.py#L54-L102

Reply:
you mean to call with arguments?